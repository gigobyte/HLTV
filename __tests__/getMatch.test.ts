import HLTV from '../src/'

function sleep(ms: number): Promise<void> {
  return new Promise(resolve => setTimeout(resolve, ms))
}

test('getMatch', async () => {
  expect(await HLTV.getMatch({ id: 2325765 })).toMatchSnapshot()
  expect(await HLTV.getMatch({ id: 2302345 })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getMatch({ id: 2290099 })).toMatchSnapshot()
  expect(await HLTV.getMatch({ id: 2186795 })).toMatchSnapshot()
}, 30000)
