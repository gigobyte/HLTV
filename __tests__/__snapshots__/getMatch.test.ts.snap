// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMatch 1`] = `
Object {
  "additionalInfo": "* Semi-final",
  "date": 1535888100000,
  "demos": Array [
    Object {
      "link": "/download/demo/43030",
      "name": "GOTV Demo",
    },
    Object {
      "link": "https://player.twitch.tv/?video=v304885542&t=3h6m9s",
      "name": "DreamHack (Pre-Match Analysis)",
    },
    Object {
      "link": "https://player.twitch.tv/?video=v304885542&t=3h28m35s",
      "name": "DreamHack (Map 1 - Dust 2)",
    },
    Object {
      "link": "https://player.twitch.tv/?video=v304885542&t=4h17m45s",
      "name": "DreamHack (Map 2 - Inferno)",
    },
    Object {
      "link": "https://player.twitch.tv/?video=v304885542&t=5h34m10s",
      "name": "DreamHack (Map 3 - Mirage)",
    },
    Object {
      "link": "https://player.twitch.tv/?video=v304885542&t=6h23m45s",
      "name": "DreamHack (Winning Moment)",
    },
    Object {
      "link": "https://player.twitch.tv/?video=v304885542&t=6h26m32s",
      "name": "DreamHack (Post-Match Analysis)",
    },
  ],
  "event": Object {
    "id": 3389,
    "name": "DreamHack Masters Stockholm 2018",
  },
  "format": "Best of 3",
  "hasScorebot": false,
  "headToHead": Array [
    Object {
      "date": 1527526761000,
      "event": Object {
        "id": 3666,
        "name": "StarSeries i-League Season 5",
      },
      "map": "trn",
      "result": "16 - 12",
      "winner": Object {
        "id": 4494,
        "name": "mousesports",
      },
    },
    Object {
      "date": 1527526761000,
      "event": Object {
        "id": 3666,
        "name": "StarSeries i-League Season 5",
      },
      "map": "mrg",
      "result": "7 - 16",
      "winner": Object {
        "id": 7533,
        "name": "North",
      },
    },
    Object {
      "date": 1527526761000,
      "event": Object {
        "id": 3666,
        "name": "StarSeries i-League Season 5",
      },
      "map": "d2",
      "result": "19 - 16",
      "winner": Object {
        "id": 4494,
        "name": "mousesports",
      },
    },
    Object {
      "date": 1520359509000,
      "event": Object {
        "id": 3361,
        "name": "ESL Pro League Season 7 Europe",
      },
      "map": "trn",
      "result": "13 - 16",
      "winner": Object {
        "id": 7533,
        "name": "North",
      },
    },
    Object {
      "date": 1520355703000,
      "event": Object {
        "id": 3361,
        "name": "ESL Pro League Season 7 Europe",
      },
      "map": "inf",
      "result": "16 - 8",
      "winner": Object {
        "id": 4494,
        "name": "mousesports",
      },
    },
    Object {
      "date": 1506461343000,
      "event": Object {
        "id": 2866,
        "name": "ESL Pro League Season 6 Europe",
      },
      "map": "mrg",
      "result": "14 - 16",
      "winner": Object {
        "id": 7533,
        "name": "North",
      },
    },
    Object {
      "date": 1506457458000,
      "event": Object {
        "id": 2866,
        "name": "ESL Pro League Season 6 Europe",
      },
      "map": "cbl",
      "result": "12 - 16",
      "winner": Object {
        "id": 7533,
        "name": "North",
      },
    },
  ],
  "highlightedPlayer": Object {
    "id": 11816,
    "name": "ropz",
  },
  "highlights": Array [
    Object {
      "link": "https://clips.twitch.tv/embed?clip=AgreeableEagerGerbilBatChest&autoplay=true",
      "title": "suNny triple entry frag (Dust2)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=AgitatedEncouragingClipsdadCurseLit&autoplay=true",
      "title": "Snax 1v2 clutch (Dust2) ",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=RoundTawdryHamsterWTRuck&autoplay=true",
      "title": "chrisJ 1v2 clutch (Dust2)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=CuteSpicyMonitorWow&autoplay=true",
      "title": "MSL triple kill hold (Inferno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=PiercingExuberantDunlinLitFam&autoplay=true",
      "title": "niko defends the with a 3k (Inferno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=ClearExpensiveQueleaKreygasm&autoplay=true",
      "title": "MSL snipes down three on the defense (Inferno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=SquareSmellyZucchiniPRChase&autoplay=true",
      "title": "suNny taps down three in the second pistol round (Inferno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=FancyInventiveBaboonNomNom&autoplay=true",
      "title": "valde takes down three to steal the round (Inferno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=ArbitraryClumsyJackalJebaited&autoplay=true",
      "title": "chrisJ scouts down three (Inferno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=ObeseMistyKangarooOMGScoots&autoplay=true",
      "title": "oskar snipes down three to keep mouz alive (Infeno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=CalmSneakySquidEagleEye&autoplay=true",
      "title": "niko 3k to steal the map (Inferno)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=HomelyOnerousBeaverGivePLZ&autoplay=true",
      "title": "aizy triple kill hold (Mirage)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=AnnoyingAbstemiousWebSmoocherZ&autoplay=true",
      "title": "Kjaerbye quick double to steal the round (Mirage)",
    },
    Object {
      "link": "https://clips.twitch.tv/embed?clip=FancyCulturedDolphinYouDontSay&autoplay=true",
      "title": "MSL triple kill hold to earn map point (Mirage)",
    },
  ],
  "id": 2325765,
  "live": false,
  "maps": Array [
    Object {
      "name": "d2",
      "result": "16:0 (15:0; 1:0)",
      "statsId": 73147,
    },
    Object {
      "name": "inf",
      "result": "14:16 (5:10; 9:6)",
      "statsId": 73164,
    },
    Object {
      "name": "mrg",
      "result": "12:16 (7:8; 5:8)",
      "statsId": 73165,
    },
  ],
  "odds": Array [],
  "oddsCommunity": undefined,
  "players": Object {
    "team1": Array [
      Object {
        "id": 798,
        "name": "oskar",
      },
      Object {
        "id": 2730,
        "name": "chrisJ",
      },
      Object {
        "id": 11816,
        "name": "ropz",
      },
      Object {
        "id": 5479,
        "name": "suNny",
      },
      Object {
        "id": 2553,
        "name": "Snax",
      },
    ],
    "team2": Array [
      Object {
        "id": 7156,
        "name": "MSL",
      },
      Object {
        "id": 8095,
        "name": "aizy",
      },
      Object {
        "id": 9031,
        "name": "valde",
      },
      Object {
        "id": 8394,
        "name": "Kjaerbye",
      },
      Object {
        "id": 10264,
        "name": "niko",
      },
    ],
  },
  "status": "Match over",
  "streams": Array [],
  "team1": Object {
    "id": 4494,
    "name": "mousesports",
  },
  "team2": Object {
    "id": 7533,
    "name": "North",
  },
  "title": undefined,
  "vetoes": Array [
    Object {
      "map": "ovp",
      "team": Object {
        "id": 4494,
        "name": "mousesports",
      },
      "type": "removed",
    },
    Object {
      "map": "cch",
      "team": Object {
        "id": 7533,
        "name": "North",
      },
      "type": "removed",
    },
    Object {
      "map": "d2",
      "team": Object {
        "id": 4494,
        "name": "mousesports",
      },
      "type": "picked",
    },
    Object {
      "map": "inf",
      "team": Object {
        "id": 7533,
        "name": "North",
      },
      "type": "picked",
    },
    Object {
      "map": "trn",
      "team": Object {
        "id": 4494,
        "name": "mousesports",
      },
      "type": "removed",
    },
    Object {
      "map": "nuke",
      "team": Object {
        "id": 7533,
        "name": "North",
      },
      "type": "removed",
    },
  ],
  "winnerTeam": Object {
    "id": 7533,
    "name": "North",
  },
}
`;

exports[`getMatch 2`] = `
Object {
  "additionalInfo": "* Semi-final",
  "date": 1461538800000,
  "demos": Array [
    Object {
      "link": "/download/demo/21006",
      "name": "GOTV Demo",
    },
    Object {
      "link": "https://embed.azubu.tv/video273395SVefragtv?t=3h27m15s&autoplay=true",
      "name": "E-Frag (Map 1 - Cobblestone)",
    },
    Object {
      "link": "https://embed.azubu.tv/video273395SVefragtv?t=4h59m0s&autoplay=true",
      "name": "E-Frag (Map 2 - Mirage)",
    },
    Object {
      "link": "https://embed.azubu.tv/video273395SVefragtv?t=5h52m45s&autoplay=true",
      "name": "E-Frag (Map 3 - Cache)",
    },
  ],
  "event": Object {
    "id": 2226,
    "name": "Rising Stars South America Promotional Cup",
  },
  "format": "Best of 3",
  "hasScorebot": false,
  "headToHead": Array [],
  "highlightedPlayer": Object {
    "id": 10499,
    "name": "JonY%20BoY",
  },
  "highlights": Array [],
  "id": 2302345,
  "live": false,
  "maps": Array [
    Object {
      "name": "cbl",
      "result": "16:6 (11:4; 5:2)",
      "statsId": 30144,
    },
    Object {
      "name": "mrg",
      "result": "13:16 (7:8; 6:8)",
      "statsId": 30145,
    },
    Object {
      "name": "cch",
      "result": "14:16 (6:9; 8:7)",
      "statsId": 30146,
    },
  ],
  "odds": Array [],
  "oddsCommunity": undefined,
  "players": Object {
    "team1": Array [
      Object {
        "id": 10559,
        "name": "knight",
      },
      Object {
        "id": 10819,
        "name": "tt",
      },
      Object {
        "id": 10818,
        "name": "iDk",
      },
      Object {
        "id": 10566,
        "name": "chelo",
      },
      Object {
        "id": 11155,
        "name": "Fluyr",
      },
    ],
    "team2": Array [
      Object {
        "id": 10499,
        "name": "JonY BoY",
      },
      Object {
        "id": 10508,
        "name": "nbl",
      },
      Object {
        "id": 11149,
        "name": "nch",
      },
      Object {
        "id": 10500,
        "name": "sheep",
      },
      Object {
        "id": 11148,
        "name": "coMplot",
      },
    ],
  },
  "status": "Match over",
  "streams": Array [],
  "team1": Object {
    "id": 6785,
    "name": "Innova",
  },
  "team2": Object {
    "id": 6798,
    "name": "2KILL",
  },
  "title": undefined,
  "vetoes": Array [],
  "winnerTeam": Object {
    "id": 6798,
    "name": "2KILL",
  },
}
`;

exports[`getMatch 3`] = `
Object {
  "additionalInfo": "",
  "date": 1392668100000,
  "demos": Array [],
  "event": Object {
    "id": 1315,
    "name": "ESL Pro Series Germany Spring 2014 Cup #1",
  },
  "format": "",
  "hasScorebot": false,
  "headToHead": Array [],
  "highlightedPlayer": undefined,
  "highlights": Array [],
  "id": 2290099,
  "live": false,
  "maps": Array [
    Object {
      "name": "nuke",
      "result": "",
      "statsId": undefined,
    },
  ],
  "odds": Array [],
  "oddsCommunity": undefined,
  "players": Object {
    "team1": Array [
      Object {
        "id": 5794,
        "name": "tabseN",
      },
      Object {
        "id": 7964,
        "name": "cadiaN",
      },
      Object {
        "id": 5796,
        "name": "tiziaN",
      },
      Object {
        "id": 2730,
        "name": "chrisJ",
      },
      Object {
        "id": 3252,
        "name": "LEGIJA",
      },
    ],
    "team2": Array [
      Object {
        "id": 7499,
        "name": "Spiidi",
      },
      Object {
        "id": 2326,
        "name": "xall",
      },
      Object {
        "id": 7246,
        "name": "stavros",
      },
      Object {
        "id": 7208,
        "name": "smn",
      },
      Object {
        "id": 7214,
        "name": "strux1",
      },
    ],
  },
  "status": "Match over",
  "streams": Array [],
  "team1": Object {
    "id": 4494,
    "name": "mousesports",
  },
  "team2": Object {
    "id": 4901,
    "name": "3DMAX",
  },
  "title": undefined,
  "vetoes": Array [],
  "winnerTeam": Object {
    "id": 4494,
    "name": "mousesports",
  },
}
`;

exports[`getMatch 4`] = `
Object {
  "additionalInfo": "",
  "date": 1352228400000,
  "demos": Array [],
  "event": Object {
    "id": 982,
    "name": "ESL Pro Series Germany Winter Season 2012",
  },
  "format": "",
  "hasScorebot": false,
  "headToHead": Array [],
  "highlightedPlayer": Object {
    "id": 7226,
    "name": "fl0w",
  },
  "highlights": Array [],
  "id": 2186795,
  "live": false,
  "maps": Array [
    Object {
      "name": "nuke",
      "result": "",
      "statsId": undefined,
    },
  ],
  "odds": Array [],
  "oddsCommunity": undefined,
  "players": Object {
    "team1": Array [
      Object {
        "id": 7224,
        "name": "caLipo",
      },
      Object {
        "id": 7225,
        "name": "cavy",
      },
      Object {
        "id": 7226,
        "name": "fl0w",
      },
      Object {
        "id": 7227,
        "name": "kev1n",
      },
      Object {
        "id": 7321,
        "name": "TuGuX",
      },
    ],
    "team2": Array [
      Object {
        "id": 7209,
        "name": "H1T",
      },
      Object {
        "id": 7212,
        "name": "miLo",
      },
      Object {
        "id": 7213,
        "name": "wNe",
      },
      Object {
        "id": 7258,
        "name": "tomo",
      },
      Object {
        "id": 7342,
        "name": "constarr",
      },
    ],
  },
  "status": "Match over",
  "streams": Array [],
  "team1": Object {
    "id": 4498,
    "name": "Red Dead",
  },
  "team2": Object {
    "id": 4499,
    "name": "CPLAY",
  },
  "title": undefined,
  "vetoes": Array [],
  "winnerTeam": Object {
    "id": 4498,
    "name": "Red Dead",
  },
}
`;
